src/main/java/hudson/tasks/test/TestResultParser.java

```diff
@@ -32,37 +32,29 @@
 import hudson.model.AbstractBuild;
 import hudson.model.Run;
 import hudson.model.TaskListener;
-import hudson.tasks.Publisher;
 
 import java.io.IOException;
 import javax.annotation.Nonnull;
 
 /**
  * Parses test result files and builds in-memory representation of it as {@link TestResult}.
- *
- * <p>
- * This extension point encapsulates the knowledge of a particular test report format and its parsing process,
- * thereby improving the pluggability of test result parsing; integration with a new test tool can be done
- * by just writing a parser, without writing a custom {@link Publisher}, and the test reports are displayed
- * with the default UI and recognized by the rest of Hudson as test reports.
- *
  * <p>
- * Most typical implementations of this class should extend from {@link DefaultTestResultParserImpl},
- * which handles a set of default error checks on user inputs. 
- *
+ * Originally this was treated as an {@link ExtensionPoint},
+ * but no supported code path actually uses this API directly. See {@link #all}.
  * <p>
  * Parsers are stateless, and the {@link #parseResult} method
  * can be concurrently invoked by multiple threads for different builds.
  *
  * @since 1.343
- * @see DefaultTestResultParserImpl
  */
-public abstract class TestResultParser implements ExtensionPoint {
+public abstract class TestResultParser {
```

Comment:
It won't be compatible in all cases after this change. Not sure what is the need in this change

Reply:
`ExtensionPoint` has no methods, and is not even treated as a marker interface at runtime. It is used purely for documentation, including [this list](https://jenkins.io/doc/developer/extensions/). Since `TestResultParser` is not in fact being used as an extension point, and AFAICT never was, this clears up the point of confusion.